### PR TITLE
implement client ping

### PR DESCRIPTION
**Pull request by nomyfan**
I found it was supported in server side.

### CODE REVIEW

File: src/bin/cli.rs [hunk 1]
```diff
@@ -26,6 +26,10 @@ struct Cli {
 
 #[derive(Subcommand, Debug)]
 enum Command {
+    Ping {
+        /// Message to ping
+        message: Option<String>,
```
Comment (hds) [inline src/bin/cli.rs:31]: Would it make sense to make this struct match the one in cmd and rename this field to `msg`?

File: src/client.rs [hunk 2]
```diff
@@ -87,6 +87,41 @@ pub async fn connect<T: ToSocketAddrs>(addr: T) -> crate::Result<Client> {
...
 
 impl Client {
+    /// Ping to the server.
+    ///
+    /// Returns PONG if no argument is provided, otherwise
+    /// return a copy of the argument as a bulk.
+    ///
+    /// This command is often used to test if a connection
+    /// is still alive, or to measure latency.
+    /// # Examples
```
Comment (hds) [inline src/client.rs:97]: Missing empty line before the Example heading.

File: tests/client.rs [hunk 3]
```diff
@@ -3,6 +3,18 @@ use std::net::SocketAddr;
...
 
+/// A basic PING PONG test.
+#[tokio::test]
+async fn ping_pong() {
+    let (addr, _) = start_server().await;
+    let mut client = client::connect(addr).await.unwrap();
+    let pong = client.ping(None).await.unwrap();
+    assert_eq!(b"PONG", &pong[..]);
+
+    let pong = client.ping(Some("你好世界".to_string())).await.unwrap();
```
Comment (hds) [inline tests/client.rs:14]: nit: I would go with keeping each test case test one thing and split this test case into 2: one for an empty message and another for a non-empty message.

**Review by hds — CHANGES_REQUESTED** [review]
A few small things to keep the code aligned with what was already there.

**Comment by nomyfan** [comment]
Changes were made.

**Review by hds — APPROVED** [review]
Thanks!

**Review by hds — APPROVED** [review]

**Review by Darksonn — APPROVED** [review]

**Pull request merged**